string.c

```diff
@@ -7314,7 +7314,8 @@ mapping_buffer_free(void *p)
 
 static const rb_data_type_t mapping_buffer_type = {
     "mapping_buffer",
-    {0, mapping_buffer_free,}
+    {0, mapping_buffer_free,},
+    0, 0, RUBY_TYPED_WB_PROTECTED
```

Comment:
Can we also add the `RUBY_TYPED_FREE_IMMEDIATELY` flag so it doesn't run as a finalizer?

Reply:
Done.